dev-util/nvidia-cuda-toolkit/nvidia-cuda-toolkit-12.8.0.ebuild

```
@@ -0,0 +1,373 @@
+# Copyright 1999-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# shellcheck disable=SC2317
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..13} )
+inherit check-reqs toolchain-funcs
+inherit python-r1
+
+DRIVER_PV="570.86.10"
+GCC_MAX_VER="14"
+CLANG_MAX_VER="19"
+
+DESCRIPTION="NVIDIA CUDA Toolkit (compiler and friends)"
+HOMEPAGE="https://developer.nvidia.com/cuda-zone"
+SRC_URI="
+	amd64? (
+		https://developer.download.nvidia.com/compute/cuda/${PV}/local_installers/cuda_${PV}_${DRIVER_PV}_linux.run
+	)
+	arm64? (
+		https://developer.download.nvidia.com/compute/cuda/${PV}/local_installers/cuda_${PV}_${DRIVER_PV}_linux_sbsa.run
+	)
+"
+S="${WORKDIR}"
+
+LICENSE="NVIDIA-CUDA"
+
+SLOT="0/${PV}" # UNSLOTTED
+# SLOT="${PV}" # SLOTTED
+
+KEYWORDS="-* ~amd64 ~arm64 ~amd64-linux ~arm64-linux"
+IUSE="debugger examples profiler rdma sanitizer"
+RESTRICT="bindist mirror strip test"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+# since CUDA 11, the bundled toolkit driver (== ${DRIVER_PV}) and the
+# actual required minimum driver version are different.
+RDEPEND="
+	|| (
+		<sys-devel/gcc-$(( GCC_MAX_VER + 1 ))_pre[cxx]
+		<llvm-core/clang-$(( CLANG_MAX_VER + 1 ))_pre
+	)
+	sys-process/numactl
+	debugger? (
+		${PYTHON_DEPS}
+	)
+	examples? (
+		media-libs/freeglut
+		media-libs/glu
+	)
+	rdma? ( sys-cluster/rdma-core )
+"
+BDEPEND="
+	$(python_gen_any_dep '
+		dev-python/defusedxml[${PYTHON_USEDEP}]
+	')
+"
+
+CUDA_PATH="/opt/cuda-${PV}"
+QA_PREBUILT="${CUDA_PATH#/}/*"
+
+python_check_deps() {
+	python_has_version "dev-python/defusedxml[${PYTHON_USEDEP}]"
+}
+
+cuda-toolkit_check_reqs() {
+	if use amd64; then
+		export CHECKREQS_DISK_BUILD="6645M"
+	elif use arm64; then
+		export CHECKREQS_DISK_BUILD="6412M"
+	fi
+
+	"check-reqs_pkg_${EBUILD_PHASE}"
+}
+
+cuda_verify() {
+	local DRIVER_PV_info
+	DRIVER_PV_info="$(sh "${DISTDIR}/${A}" --info | grep -oP "cuda_${PV}.*run" | cut -d '_' -f 3)"
+
+	if [[ "${DRIVER_PV}" != "${DRIVER_PV_info}" ]]; then
+		die "check DRIVER_PV is ${DRIVER_PV} and should be ${DRIVER_PV_info}"
+	fi
+
+	# rest only works in with unpacked sources
+	[[ "${EBUILD_PHASE}" != prepare ]] && return
+
+	# run self checks
+	local compiler_versions GCC_HAS_VER CLANG_HAS_VER
+	compiler_versions="$(
+		grep -oP "unsupported (GNU|clang) version.*(gcc versions later than|clang version must be less than) [0-9]*" \
+			"${S}"/builds/cuda_nvcc/targets/*/include/crt/host_config.h
+	)"
+
+	GCC_HAS_VER="$(      echo "${compiler_versions}" | grep   gcc | grep -oP "(?<=than )[0-9]*")"
+	if [[   "${GCC_MAX_VER}" -ne   "${GCC_HAS_VER}" ]]; then
```

No alignment like this, please:
```suggestion
	if [[ "${GCC_MAX_VER}" -ne "${GCC_HAS_VER}" ]]; then
```

See https://projects.gentoo.org/qa/policy-guide/ebuild-format.html#ebuild-file-format.

---

This is technically not indenting, but so be it.